lute/static/js/dict-tabs.js

```diff
@@ -0,0 +1,455 @@
+  const imageBtn = createTabBtn("", dictTabsContainer, -1, 0);
+  imageBtn.setAttribute("id", "dict-image-btn");
+  imageBtn.setAttribute("title", "Look up images for the term");
+  const imageFrame = createIFrame("imageframe", iFramesContainer);
+
+  dictTabButtons.set(imageBtn, imageFrame);
+  // create sentences frame
+  const sentencesFrame = createIFrame("sentencesframe", iFramesContainer);
+  dictTabButtons.set("sentencesTab", sentencesFrame);
+
+  dictTabsContainer.addEventListener("click", (e) => {
+    tabsClick(e, dictTabButtons);
+  });
+
+  return dictTabButtons;
+}
+
+function tabsClick(event, dictTabButtons) {
+  const clickedTab = event.target.closest(".dict-btn");
+  if (!clickedTab) return;
+
+  const isExternal = clickedTab.dataset.dictExternal == 1 ? true : false;
+  const dictID = clickedTab.dataset.dictId;
+
+  if (isExternal) {
+    loadDictPage(dictID, "");
+  } else {
+    const iFrame = dictTabButtons.get(clickedTab);
+    if (iFrame.dataset.tabOpened == 0) {
+      loadDictPage(dictID, iFrame);
+    }
+    iFrame.dataset.tabOpened = 1;
+    activateTab(clickedTab, dictTabButtons);
+  }
+}
+
+function listMenuClick(event, listMenuContainer, btn, dictTabButtons, iFrame) {
+  const clickedOption = event.target.closest(".dict-select-option");
+  if (!clickedOption) return;
+
+  listMenuContainer.classList.add("dict-select-list-hide");
+
+  const optionVal = clickedOption.dataset.dictId;
+  const domain = getDictURLDomain(TERM_DICTS[optionVal]);
+  const btnLabel = domain.split("www.").splice(-1)[0];
+  const faviconURL = getFavicon(domain);
+  const faviconEl = createImg(faviconURL, "dict-btn-fav-img"); // img elements get deleted after "change" event. so we create them after each change
+  
+  btn.dataset.dictId = optionVal;
+  btn.dataset.dictExternal = clickedOption.dataset.dictExternal;
+  btn.textContent = btnLabel;
+  // btn.dataset.tabOpened = clickedOption.dataset.tabOpened;
+  btn.prepend(faviconEl);
+  const menuImgEl = createImg("", "dict-btn-list-img");
+  btn.appendChild(menuImgEl);
+  
+  if (clickedOption.dataset.dictExternal == 1) {
+    loadDictPage(optionVal, "");
+
+    const arrowEl = createImg("", "dict-btn-external-img");
+    btn.appendChild(arrowEl);
+    // btn.classList.remove("dict-btn-active");
+  } else {
+    loadDictPage(optionVal, iFrame);
+    activateTab(btn, dictTabButtons);
+  }
+  // as with the icons, btn content changes so events get deleted
+  menuImgEl.addEventListener("click", (e) => {
+    e.stopPropagation();
+    listMenuContainer.classList.toggle("dict-select-list-hide");
+  });
+}
+
+function createDictListMenu(dicts) {
+  const listContainer = document.createElement("div");
+  listContainer.setAttribute("id", "dict-select-container");
+  listContainer.classList.add("dict-select-list-hide");
+  
+  dicts.forEach((dict) => {
+    const menuItem = document.createElement("p");
+    menuItem.classList.add("dict-select-option");
+    const origIndex = TERM_DICTS.indexOf(dict);
```

Comment:
Code tip: this line is very short, and the var `origIndex` is only used once, so you could skip creating a var and just assign it later, like `menuItem.dataset.dictId = TERM_DICTS.indexOf(dict);`.  It's still very clear what's up.  Same goes for other vars throughout. :-)